schema/schema.go

```diff
@@ -346,6 +348,62 @@ func (schema *Schema) StateVersioning() bool {
 	return stateful
 }
 
+func (schema *Schema) SyncKeyTemplate() string {
+	syncKeyTemplateRaw, ok := schema.Metadata["sync_key_template"]
+	if !ok {
+		return ""
+	}
+	syncKeyTemplate, ok := syncKeyTemplateRaw.(string)
+	if !ok {
+		return ""
+	}
+	return syncKeyTemplate
+}
+
+func (schema *Schema) GenerateCustomPath(data map[string]interface{}) (string, error) {
+	tpl, err := pongo2.FromString(schema.SyncKeyTemplate())
+	if err != nil {
+		return "", err
+	}
+	path, err := tpl.Execute(pongo2.Context{}.Update(data))
+	if err != nil {
+		return "", err
+	}
+	return path, nil
+}
+
+func (schema *Schema) GetResourceIdFromPath(schemaPath string) string {
+	resourceID := ""
+	if schema.SyncKeyTemplate() == "" {
```

Comment:
[#comment](https://github.com/cloudwan/gohan/pull/227/files#r64607353) ... and using it here.